pump/server.go

```diff
@@ -692,77 +685,38 @@ func (s *Server) waitSafeToOffline(ctx context.Context) error {
 	var fakeBinlog *pb.Binlog
 	var err error
 
-	// write a fake binlog
-	for {
+	err = util.TryUntilSuccess(ctx, time.Second, "Failed to write fake binlog", func() error {
 		fakeBinlog, err = s.writeFakeBinlog()
```

Comment:
i think this `err` will capture the `err` at L686?
will it clear to declare a new `err` inside the annoy func?